src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -187,9 +185,14 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
 
         if ('' === $url) {
             $url = '/';
-        } elseif (preg_match($this->urlEncodingSkipRegexp, $url)) {
-            // the context base URL is already encoded (see Symfony\Component\HttpFoundation\Request)
+        } elseif (!empty($this->decodedChars)) {
+            @trigger_error('The class variable '.__CLASS__.'::$decodedChars is deprecated since version 2.7 and will be removed in 3.1.', E_USER_DEPRECATED);
```

Comment:
@nicolas-grekas while I appreciate your suggestion and it would allow early inclusion of this change. Its also not in the spirit of the intended prevention of BC-breakage. If someone would rely on the contents of the decodedChars-array, that would still break (i.e. it would be an empty array, rather than one containing several key/value pairs) and without any warning. Its already a bug waiting to happen in its current state; changing the array without changing the url matcher could result in mismatched routes...

Adding more code to test if the array was non-default, could destroy most of the performance gain.

So I'll rebase it against master, where such a change is much less unexpected. So I'll follow the suggestion of @iltar